addon/components/g-map.js

```diff
@@ -25,6 +26,11 @@ export default class GMap extends MapComponent {
 
   components = new Set();
 
+  constructor(owner, args, options, events) {
+    super(owner, args, options, events);
+    this.options.mapId = uuidv4();
```

Comment:
A unique `mapId` needs to be set to be able to use `advanced-marker`s

Reply:
I had put it in the `newOptions` getter but while running tests, I was getting a warning that `mapId` shouldn't change. What do you think?

Reply:
Then I would generate the uuid once (right on the class) and then do something like:

```javascript
this.options.mapId ??= this.mapId
```

Could generate it on-demand and cache it, but it's not a big deal. It's not like anyone is creating thousands of maps on one page.